src/platform/plugins/shared/esql/public/triggers/esql_controls/control_flyout/index.tsx

```diff
@@ -154,16 +155,21 @@ export function ESQLControlsFlyout({
   }, []);
 
   const onCreateControl = useCallback(async () => {
-    if (controlState && controlState.availableOptions.length) {
+    if (controlState && controlState.availableOptions?.length) {
```

Comment:
how about moving controlState.availableOptions logic into src/platform/plugins/shared/controls/public/controls/esql_control/get_esql_control_factory.tsx `onSaveControl` so serialize implementation details do not leak into the UI.